src/eligibility-check-opkit.ts

```diff
@@ -0,0 +1,315 @@
+  // Because this bot is triggered by a subscription, the resource that comes in is a Coverage object
+  const coverage = event.input as Coverage;
+  const patient = await medplum.readReference(coverage.subscriber as Reference<Patient>);
+  const organization: Organization = await medplum.readReference(coverage.payor?.[0] as Reference<Organization>);
+  const provider: Practitioner = await medplum.readReference(
+    patient.generalPractitioner?.[0] as Reference<Practitioner>
+  );
+
+  if (!coverage) {
+    console.log('No coverage found');
+    return true;
+  }
+
+  if (!patient) {
+    console.log('No patient found');
+    return true;
+  }
+
+  if (!organization) {
+    console.log('No payor found');
+    return true;
+  }
+
+  if (!provider) {
+    console.log('No provider found');
+    return true;
+  }
+
+  var coverageEligibilityReq: CoverageEligibilityRequest = await medplum.createResource({
+    resourceType: 'CoverageEligibilityRequest',
+    provider: createReference(provider),
+    patient: createReference(patient),
+    insurer: createReference(organization),
+    insurance: [createReference(coverage)],
+  });
+
+  const providerNpi = provider.identifier?.find(
+    (identifier) => identifier.system === 'http://hl7.org/fhir/sid/us-npi'
+  )?.value;
+  const serviceTypes = ['health_benefit_plan_coverage'];
+  const opkitKey = '<opkit-public-api-key>';
+  const payerId = organization.identifier?.find(
+    (identifier) => identifier.system === 'https://docs.opkit.co/reference/getpayers'
+  )?.value;
+  const patientEmail = patient.telecom?.find((identifier) => identifier.system === 'email')?.value;
+
+  const opkitRequest = {
+    provider_npi: providerNpi,
+    payer_id: payerId,
+    subscriber: {
+      first_name: patient.name?.[0]?.given?.[0],
+      last_name: patient.name?.[0]?.family,
+      member_id: coverage.subscriberId,
+      date_of_birth: patient.birthDate,
+      email: patientEmail,
+    },
+    services: serviceTypes,
+  };
+
+  const result: any = await fetch('https://api.opkit.co/v1/eligibility_inquiries', {
+    method: 'POST',
+    body: JSON.stringify(opkitRequest),
+    headers: {
+      Authorization: 'Basic ' + Buffer.from(`${opkitKey}:`).toString('base64'),
+      'Content-Type': 'application/json',
+      Accept: 'application/json',
+    },
+  })
+    .then((response) => response.json())
+    .catch((error) => console.log('Error checking eligibility request'));
+
+  if (!result || !result?.plan?.benefits) {
+    return false;
+  }
+
+  const serviceTypeCodeToCategoryDisplayMap: Record<string, string> = {
+    health_benefit_plan_coverage: 'Health Benefit Plan Coverage',
+  };
+
+  const serviceTypeCodeToCategoryCodeMap: Record<string, string> = {
+    health_benefit_plan_coverage: '30',
+  };
+
+  const networkToNetworkTypeCodeMap: Record<string, string> = {
+    out_of_network: 'out',
+    in_network: 'in',
+  };
+
+  const coverageToBenefitUnitMap: Record<string, string> = {
```

Comment:
Down the road, it feels like all these maps are probably good candidates for the FHIR Mapping Language functionality (or ConceptMaps, I'm not sure which)